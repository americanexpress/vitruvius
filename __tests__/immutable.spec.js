/*
 * Copyright (c) 2017 American Express Travel Related Services Company, Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
 * in compliance with the License. You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under the License
 * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
 * or implied. See the License for the specific language governing permissions and limitations under
 * the License.
 */

import vitruviusImmutable from '../src/immutable';
import vitruviusImmutableExport from '../immutable';
import { runVitruviusTests } from './fixtures';

// eslint-disable-next-line global-require  -- no context
jest.mock('../lib/immutable', () => require('../src/immutable'), { virtual: true });

runVitruviusTests('vitruviusImmutable', vitruviusImmutable);

describe('vitruviusImmutable export', () => {
  it('should export vitruviusImmutable', () => {
    expect(vitruviusImmutableExport.name).toBe('vitruviusImmutable');
  });
});
