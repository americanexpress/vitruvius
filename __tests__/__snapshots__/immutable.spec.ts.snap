// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`vitruviusImmutable > should build the expected state for a flat tree of reducers 1`] = `
Immutable.Map {
  "stuff": Immutable.Map {
    "foo": "test",
  },
  "things": Immutable.Map {
    "bar": "test",
  },
  "other": Immutable.Map {
    "baz": "test",
  },
}
`;

exports[`vitruviusImmutable > should build the expected state for a nested tree of reducers 1`] = `
Immutable.Map {
  "stuff": Immutable.Map {
    "foo": "test",
  },
  "nested": Immutable.Map {
    "things": Immutable.Map {
      "bar": "test",
    },
    "other": Immutable.Map {
      "baz": "test",
    },
  },
}
`;

exports[`vitruviusImmutable > should handle a mix of reducers having and not having buildInitialState method 1`] = `
Immutable.Map {
  "stuff": Immutable.Map {
    "foo": "test",
  },
  "things": Immutable.Map {
    "bar": "test",
  },
  "other": Immutable.Map {
    "baz": "test",
  },
}
`;

exports[`vitruviusImmutable > should handle all reducers having buildInitialState method 1`] = `
Immutable.Map {
  "stuff": Immutable.Map {
    "foo": "test",
  },
  "things": Immutable.Map {
    "bar": "test",
  },
  "other": Immutable.Map {
    "baz": "test",
  },
}
`;

exports[`vitruviusImmutable > should handle no reducers having buildInitialState method 1`] = `Immutable.Map {}`;

exports[`vitruviusImmutable > should return an initialState that is acceptable to redux's createStore 1`] = `
Immutable.Map {
  "stuff": Immutable.Map {
    "foo": "test",
  },
  "things": Immutable.Map {
    "bar": "test",
  },
  "other": Immutable.Map {
    "baz": "test",
  },
  "static": Immutable.Map {
    "static": true,
  },
}
`;
